pkg/identityserver/store/application_store_test.go

```diff
@@ -128,5 +128,16 @@ func TestApplicationStore(t *testing.T) {
 
 		a.So(err, should.BeNil)
 		a.So(list, should.BeEmpty)
+
+		err = store.PurgeApplication(ctx, &ttnpb.ApplicationIdentifiers{ApplicationID: "foo"})
+
+		a.So(err, should.BeNil)
+
+		// Check that application ids are released after purge
+		_, err = store.CreateApplication(ctx, &ttnpb.Application{
+			ApplicationIdentifiers: ttnpb.ApplicationIdentifiers{ApplicationID: "foo"},
+		})
+
+		a.So(err, should.BeNil)
```

Comment:
You might also want to check whether associated API Keys and/or memberships were deleted.

Reply:
The issue with testing this is that it's not part of `Purge` application store method. It is a part of the `Purge` method in the application registry. I think what is ensured with the separate store functions is that with the same input, the behavior is correct. In the store it makes sense to ensure that the fields like `Picture`, `Attribute`, and `Antenna` are cleared properly because that is not covered with tests. Not sure if having the functionality to test the full flow should reside in the unit tests.